moveit_ros/moveit_servo/src/servo_calcs.cpp

```diff
@@ -317,38 +301,68 @@ void ServoCalcs::stop()
 
 void ServoCalcs::mainCalcLoop()
 {
-  rclcpp::WallRate rate(1.0 / parameters_->publish_period);
+  rclcpp::WallRate rate(1.0 / servo_params_.publish_period);
 
   while (rclcpp::ok() && !stop_requested_)
   {
     // lock the input state mutex
     std::unique_lock<std::mutex> main_loop_lock(main_loop_mutex_);
 
     // low latency mode -- begin calculations as soon as a new command is received.
-    if (parameters_->low_latency_mode)
+    if (servo_params_.low_latency_mode)
     {
       input_cv_.wait(main_loop_lock, [this] { return (new_input_cmd_ || stop_requested_); });
     }
 
     // reset new_input_cmd_ flag
     new_input_cmd_ = false;
 
+    // Check if any parameters changed
+    // TODO : Remove debug info here
+    if (servo_param_listener_->is_old(servo_params_))
```

Comment:
To answer your top-level question about what to do about parameters: Many other nodes that use this library basically have a boolean parameter that configures whether the node should be continuously checking for parameter updates on every tick vs. just initialize once and not check again. For performance reasons.

an example of this would be [the ros2_control admittance controller](https://github.com/ros-controls/ros2_controllers/blob/master/admittance_controller/src/admittance_controller_parameters.yaml#L155-L159), where the use of this parameter would be [here](https://github.com/ros-controls/ros2_controllers/blob/master/admittance_controller/include/admittance_controller/admittance_rule_impl.hpp#L169-L172).

Reply:
Thanks !
Will add that here as well .